---
title: "Project Pipeline"
author: "Sarah Bowker"
date: "2025-10-20"
output: html_document
---

# Step1:
```{bash sequali_report, message=FALSE, warning=FALSE}
# Get sequali report
conda activate sequali_env
sequali --outdir ./sequali_reports "./Lachancea thermotolerans.fastq"

# Filter the long read fastq file to keep the longest 75% of the sequesnce
conda install filtlong
filtlong -p 90 "Lachancea thermotolerans.fastq" > "Lachancea_thermotolerans_90filtered.fastq"

filtlong -p 95 "Lachancea thermotolerans.fastq" > "Lachancea_thermotolerans_95filtered.fastq"

# Run sequali report on the filtered fastq file
sequali --outdir ./sequali_reports "./Lachancea_thermotolerans_85filtered.fastq"

conda activate flye-env
flye --nano-raw Lachancea_thermotolerans_85filtered.fastq --out-dir flye_output --threads 8 --genome-size 10m

     
flye --nano-hq "Lachancea_thermotolerans_95filtered.fastq"\
     --out-dir flye_output_scaffolded \
     --threads 8 \
     --genome-size 10m \
     --scaffold \
     --iterations 5
     
flye --nano-hq Lachancea_thermotolerans_95filtered.fastq \
     --out-dir flye_output_unscaffolded \
     --threads 8 \
     --genome-size 10m \
     --iterations 5
```


# step 2:
``````{bash , eval=FALSE, message=FALSE, warning=FALSE}


docker run --rm \
  -v "$PWD":/data \
  -w /data \
  ezlabgva/busco:v6.0.0_cv1 \
  busco -i assembly_unscaffolded.fasta -o busco_output -l saccharomycetaceae_odb12 -m genome -c 8


docker run --rm -v "$(pwd)":/data -w /data quay.io/biocontainers/quast:5.3.0--py313pl5321h5ca1c30_2 \
  quast.py assembly_unscaffolded.fasta \
  -r GCF_000142805.1_ASM14280v1_genomic.fna \
  -g GCF_000142805.1_ASM14280v1_genomic.gff \
  -o quast_output
  
docker run --rm -v "$(pwd)":/data -w /data quay.io/biocontainers/quast:5.3.0--py313pl5321h5ca1c30_2 \
  quast.py assembly_scaffolded.fasta \
  -r GCF_000142805.1_ASM14280v1_genomic.fna \
  -g GCF_000142805.1_ASM14280v1_genomic.gff \
  -o quast_scaffolded_output


docker run --rm -v "$PWD":/data -w /data \
  nextgenusfs/funannotate funannotate clean \
 -i assembly_unscaffolded.fasta -o assembly_unscaffolded_cleaned.fasta

  
  
docker run --rm -v "$PWD":/data -w /data \
  nextgenusfs/funannotate funannotate clean \
 -i assembly_scaffolded.fasta -o assembly_scaffolded_cleaned.fasta
 
 

docker run --rm -v "$PWD":/data -w /data \
  nextgenusfs/funannotate funannotate mask \
 -i assembly_unscaffolded_cleaned.fasta -o assembly_unscaffolded_masked.fasta

docker run --rm -v "$PWD":/data -w /data \
  nextgenusfs/funannotate funannotate mask \
 -i assembly_scaffolded_cleaned.fasta -o assembly_scaffolded_masked.fasta  



docker run --rm -v "$PWD":/data -w /data --platform linux/amd64 \
  nextgenusfs/funannotate funannotate predict \
  -i assembly_unscaffolded_masked.fasta \
  -o /data/unscaffolded_output \
  --species "Lachancea thermotolerans" \
  --strain my_strain \
  --protein_evidence /data/GCF_000142805.1_ASM14280v1_protein.faa \
  --transcript_evidence /data/GCF_000142805.1_ASM14280v1_cds_from_genomic.fna \
  --cpus 8
  
  

docker run --rm -v "$PWD":/data -w /data --platform linux/amd64 \
  nextgenusfs/funannotate funannotate predict \
  -i assembly_scaffolded_masked.fasta \
  -o /data/scaffolded_output \
  --species "Lachancea thermotolerans" \
  --strain my_strain \
  --protein_evidence /data/GCF_000142805.1_ASM14280v1_protein.faa \
  --transcript_evidence /data/GCF_000142805.1_ASM14280v1_cds_from_genomic.fna \
  --cpus 8

  
docker run --rm -v "$PWD":/data -w /data --platform linux/amd64 \
  nextgenusfs/funannotate funannotate annotate \
    -i /data/unscaffolded_output \
    -o /data/unscaffolded_output_annotated \
    --cpus 8


docker run --rm -v "$PWD":/data -w /data --platform linux/amd64 \
  nextgenusfs/funannotate funannotate annotate \
    -i /data/scaffolded_output \
    -o /data/scaffolded_output_annotated \
    --cpus 8
```    
## Found 0 duplicated annotations, adding 12,439 valid annotations
```


```
conda activate sequali_env
sequali --outdir ./sequali_reports "./lachanceaFermentati.fastq"

filtlong -p 95 "lachanceaFermentati.fastq" > "lachanceaFermentati.fastq_95filtered.fastq"

filtlong -p 92 "lachanceaFermentati.fastq" > "lachanceaFermentati.fastq_92filtered.fastq"

sequali --outdir ./sequali_reports "./lachanceaFermentati.fastq_92filtered.fastq"

flye --nano-hq lachanceaFermentati.fastq_92filtered.fastq \
     --out-dir flye_output_lachanceaFermentati \
     --threads 8 \
     --genome-size 10m \
     --iterations 5

docker run --rm \
  -v "$PWD":/data \
  -w /data \
  ezlabgva/busco:v6.0.0_cv1 \
  busco -i assembly_lachanceaFermentati.fasta -o busco_output_lachanceaFermentati -l saccharomycetaceae_odb12 -m genome -c 8
  
  
  
docker run --rm -v "$(pwd)":/data -w /data quay.io/biocontainers/quast:5.3.0--py313pl5321h5ca1c30_2 \
  quast.py assembly_lachanceaFermentati.fasta \
  -r GCA_900074765.1_LAFE0_genomic.fna \
  -g GCA_900074765.1_LAFE0_genomic.gff \
  -o quast_output_lachanceaFermentati
  
  
  
docker run --rm -v "$PWD":/data -w /data \
  nextgenusfs/funannotate funannotate clean \
 -i assembly_lachanceaFermentati.fasta -o assembly_lachanceaFermentati_cleaned.fasta
 
 
docker run --rm -v "$PWD":/data -w /data \
  nextgenusfs/funannotate funannotate mask \
 -i assembly_lachanceaFermentati_cleaned.fasta -o assembly_lachanceaFermentati_masked.fasta
 
 
 
docker run --rm -v "$PWD":/data -w /data --platform linux/amd64 \
  nextgenusfs/funannotate funannotate predict \
  -i assembly_lachanceaFermentati_masked.fasta \
  -o /data/lachanceaFermentati_output \
  --species "Lachancea Fermentati" \
  --strain my_strain \
  --protein_evidence /data/GCA_900074765.1_LAFE0_protein.faa \
  --transcript_evidence /data/GCA_900074765.1_LAFE0_cds_from_genomic.fna \
  --cpus 8
  
  
docker run --rm -v "$PWD":/data -w /data --platform linux/amd64 \
  nextgenusfs/funannotate funannotate annotate \
    -i /data/lachanceaFermentati_output \
    -o /data/lachanceaFermentati_output_annotated \
    --cpus 8
    
# Found 0 duplicated annotations, adding 5,186 valid annotations


```